app/gui/src/project-view/stores/projectNames.ts

```diff
@@ -64,8 +78,15 @@ function useProjectNameStore(
     return path.path ? qnJoin(project, path.path) : project
   }
 
+  function pathSegments(projectPath: ProjectPath) {
```

Comment:
I'm not sure this fits in the scope of the `ProjectNameStore`. The store API exposes an important distinction between parse/print/serialize. This helper is equivalent to `qnSegments(printProjectPath(...))`, but hides which type of printing is being used. It is more convenient and potentially more efficient than using the two functions separately, but I think for those benefits to justify the cognitive overhead of expanding this API, it would need to be used a lot.